lib/dal/src/attribute/value.rs

```diff
@@ -549,93 +550,121 @@ impl AttributeValue {
         let destination_component_id =
             AttributeValue::component_id(ctx, attribute_value_id).await?;
         let value_is_for = AttributeValue::is_for(ctx, attribute_value_id).await?;
+        info!(
+            "Attribute value Id {} value is for {:?}",
+            attribute_value_id, value_is_for
+        );
         let apa_ids = AttributePrototypeArgument::list_ids_for_prototype(ctx, prototype_id).await?;
         let mut func_binding_args: HashMap<String, Vec<serde_json::Value>> = HashMap::new();
-
-        for apa_id in apa_ids {
-            let apa = AttributePrototypeArgument::get_by_id(ctx, apa_id).await?;
-            let expected_source_component_id = apa
-                .targets()
-                .map(|targets| targets.source_component_id)
-                .unwrap_or(destination_component_id);
-
-            if apa.targets().map_or(true, |targets| {
-                targets.destination_component_id == destination_component_id
-            }) {
-                // If the "source" Component is marked for deletion, and we (the destination) are
-                // *NOT*, then we should ignore the argument as data should not flow from things
-                // that are marked for deletion to ones that are not.
-                let destination_component = Component::get_by_id(ctx, destination_component_id)
-                    .await
-                    .map_err(Box::new)?;
-                let source_component = Component::get_by_id(ctx, expected_source_component_id)
-                    .await
-                    .map_err(Box::new)?;
-                if source_component.to_delete() && !destination_component.to_delete() {
-                    continue;
-                }
-
-                let func_arg_id =
-                    AttributePrototypeArgument::func_argument_id_by_id(ctx, apa_id).await?;
-                let func_arg_name = ctx
-                    .workspace_snapshot()?
-                    .get_node_weight_by_id(func_arg_id)
-                    .await?
-                    .get_func_argument_node_weight()?
-                    .name()
-                    .to_owned();
-
-                let values_for_arg =
-                    match AttributePrototypeArgument::value_source_by_id(ctx, apa_id)
-                        .await?
-                        .ok_or(
-                            AttributeValueError::AttributePrototypeArgumentMissingValueSource(
-                                apa_id,
-                            ),
-                        )? {
-                        ValueSource::StaticArgumentValue(static_argument_value_id) => {
-                            vec![
-                                StaticArgumentValue::get_by_id(ctx, static_argument_value_id)
-                                    .await?
-                                    .value,
-                            ]
+        if apa_ids.is_empty() {
```

Comment:
In the case of sockets with an Arity of many, do we want to allow both manually drawn connections and inferred connections from the frame relationships?

Reply:
Mixing implicit & explicit connections like that seems like it'll lead to a lot of complexity/confusion. I think it'd be a lot clearer for users if we say that there are either inferred connections, or manual connections, but not both on the same socket. If they would like both manual data, and what would have been coming from the inferred connection, then it's possible to make both connections manually. Having the inferred one still exist when there's a manual connection means that the inferred one can't ever be overridden.

Reply:
Yeah I think we *could* do it in the future, but not without loads more UX to make it easier to understand what is happening.